dali/operators/reader/loader/numpy_loader.cc

```diff
@@ -38,6 +38,124 @@ TypeInfo TypeFromNumpyStr(const std::string &format) {
   DALI_FAIL("Unknown Numpy type string");
 }
 
+inline void SkipSpaces(const char*& ptr) {
+  while (::isspace(*ptr))
+    ptr++;
+}
+
+template <size_t N>
+void Skip(const char*& ptr, const char (&what)[N]) {
+  DALI_ENFORCE(!strncmp(ptr, what, N - 1),
+               make_string("Expected \"", what, "\" but got \"", ptr, "\""));
+  ptr += N - 1;
+}
+
+template <size_t N>
+bool TrySkip(const char*& ptr, const char (&what)[N]) {
+  if (!strncmp(ptr, what, N-1)) {
+    ptr += N - 1;
+    return true;
+  } else {
+    return false;
+  }
+}
+
+
+template <size_t N>
+void SkipFieldName(const char*& ptr, const char (&name)[N]) {
+  SkipSpaces(ptr);
+  Skip(ptr, "'");
+  Skip(ptr, name);
+  Skip(ptr, "'");
+  SkipSpaces(ptr);
+  Skip(ptr, ":");
+  SkipSpaces(ptr);
+}
+
+template <typename T = int64_t>
+T ParseInteger(const char*& input) {
+  T value = static_cast<T>(
+    strtol(input, const_cast<char**>(&input), 10));  // why is it char** ?
+  return value;
```

Comment:
Shouldn't it throw when input has not advanced?
```suggestion
  char *out = const_cast<char *>(input);
  T value = static_cast<T>(strtol(input, &out, 10));
  DALI_ENFORCE(out != input, "Parse error: expected a number");
  input = out;
  return value;
```